packages/lodestar/src/network/gossip/topic.ts

```diff
@@ -52,19 +49,28 @@ export function getSubnetFromAttestationSubnetTopic(topic: string): number {
 /**
  * Create a `GossipTopic` from a gossip topic string
  */
-export function getGossipTopic(config: IBeaconConfig, topic: string, genesisValidatorsRoot: Root): GossipTopic {
+export function getGossipTopic(forkDigestContext: IForkDigestContext, topic: string): GossipTopic {
   const groups = topic.match(GossipTopicRegExp);
-  if (!groups || !groups[4]) {
+  if (!groups) {
+    throw Error(`Bad gossip topic string: ${topic}`);
+  }
+
+  const forkDigestHex = groups[2] as string | undefined;
+  const type = groups[3] as GossipType | undefined;
+  const encoding = groups[4] as GossipEncoding | undefined;
```

Comment:
De-structure first at once for context. Cast to T | undefined which is the real type